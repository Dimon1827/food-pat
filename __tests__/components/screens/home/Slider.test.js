import Slider from '@/components/screens/home/example/slider/Slider';
import { screen, render, fireEvent } from '@testing-library/react';
import React from 'react';



describe('Slider', () => {
  it('Slider renders correctly', () => {
    render(<Slider />);
    expect(screen.getByTestId('slider')).toMatchSnapshot();
  });
  it('Slider button should have position is 42%', () => {
    render(<Slider />);
    const sliderButton = screen.getByRole('button', {
      name: /Кнопка для изменения пропорций/,
    });

    expect(sliderButton).toHaveStyle({ left: 'calc(42% - 12px)' });
  });
  it('Slider should update button if button is down and move', () => {
    render(<Slider />);
    const sliderImages = screen.getByTestId('slider-images');
    const sliderButton = screen.getByRole('button', {
      name: /Кнопка для изменения пропорций/,
    });
    fireEvent.mouseDown(sliderImages);
    fireEvent.mouseMove(sliderImages, { client: 100 });
    fireEvent.mouseUp(sliderImages);
    expect(sliderButton).not.toHaveStyle({ left: 'calc(42% - 12px)' });
  });
  it('Slider doesnt should change a location of button, if button doesnt down', () => {
    render(<Slider />);
    const sliderImages = screen.getByTestId('slider-images');

    const setIsDragging = jest.fn();
    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [false, setIsDragging]);
    fireEvent.mouseMove(sliderImages, { client: 100 });
    expect(setIsDragging).not.toHaveBeenCalled();
  });
});
