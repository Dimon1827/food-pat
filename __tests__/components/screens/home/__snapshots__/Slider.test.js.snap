// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Slider renders correctly 1`] = `
<div
  class="slider"
  data-testid="slider"
>
  <div
    class="example__slider-images slider__images"
    data-testid="slider-images"
  >
    <div
      class="slider__image-wrapper slider__image-wrapper--before"
      style="clip-path: inset(0 58% 0 0);"
    >
      <picture>
        <source
          height="737"
          media="(min-width:1280px)"
          srcset="./assets/images/woman-before-tablet.webp"
          type="image/webp"
          width="584"
        />
        <source
          height="737"
          media="(min-width:768px)"
          srcset="./assets/images/woman-before-tablet.webp"
          type="image/webp"
          width="584"
        />
        <source
          height="480"
          srcset="./assets/images/woman-before-mobile.webp"
          type="image/webp"
          width="312"
        />
        <source
          height="737"
          media="(min-width:1280px)"
          srcset="./assets/images/woman-before-tablet.png"
          type="image/png"
          width="584"
        />
        <source
          height="737"
          media="(min-width:768px)"
          srcset="./assets/images/woman-before-tablet.png"
          type="image/png"
          width="584"
        />
        <img
          alt="Женщина до"
          class="example__image"
          draggable="false"
          height="480"
          src="./assets/images/woman-before-mobile.png"
          width="312"
        />
      </picture>
    </div>
    <button
      class="example__button"
      style="left: calc(42% - 12px);"
      type="button"
    >
      <span
        class="example__button-toogle"
      />
      <span
        class="visually-hidden"
      >
        Кнопка для изменения пропорций
      </span>
    </button>
    <div
      class="slider__image-wrapper slider__image-wrapper--after"
    >
      <picture>
        <source
          height="737"
          media="(min-width:1280px)"
          srcset="./assets/images/woman-after-tablet.webp"
          type="image/webp"
          width="584"
        />
        <source
          height="737"
          media="(min-width:768px)"
          srcset="./assets/images/woman-after-tablet.webp"
          type="image/webp"
          width="584"
        />
        <source
          height="480"
          srcset="./assets/images/woman-after-mobile.webp"
          type="image/webp"
          width="312"
        />
        <source
          height="737"
          media="(min-width:1280px)"
          srcset="./assets/images/woman-after-tablet.png"
          type="image/png"
          width="584"
        />
        <source
          height="737"
          media="(min-width:768px)"
          srcset="./assets/images/woman-after-tablet.png"
          type="image/png"
          width="584"
        />
        <img
          alt="Женщина до"
          class="example__image"
          draggable="false"
          height="480"
          src="./assets/images/woman-after-mobile.png"
          width="312"
        />
      </picture>
    </div>
  </div>
</div>
`;
